core/src/test/java/org/apache/calcite/rex/RexBuilderTest.java

```diff
@@ -414,6 +414,23 @@ private void checkDate(RexNode node) {
     assertThat(literal.getValue3(), nullValue());
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-3587">[CALCITE-3587]
+   * RexBuilder may lose decimal fraction for creating literal with DECIMAL type</a>.
+   */
```

Comment:
lose decimal fraction -> loose decimal precision ?

Reply:
no. It's lose, not loose.
Loose is an adjective that means “not tight.”

Reply:
What do you mean by "fraction" ?

Reply:
"decimal fraction", means the non integer part . for "12.95", means the ".95" part.